assets/src/components/activities/adaptive/components/authoring/LayoutEditor.tsx

```diff
@@ -148,39 +143,75 @@ const LayoutEditor: React.FC<LayoutEditorProps> = (props) => {
       if (selectedPartId === payload.id) {
         return;
       }
-      setSelectedPartId(payload.id);
       props.onSelect(payload.id);
     },
-    [selectedPartId],
+    [props.onSelect, selectedPartId],
   );
 
-  const handlePartDrag = useCallback(
-    async (payload: any) => {
-      // console.log('AUTHOR PART DRAG', payload);
-      if (payload.dragData.deltaX === 0 && payload.dragData.deltaY === 0) {
+  /**
+   * Given a part ID, this will clone and modify values on part.custom and post the changes up the chain.
+   *
+   * Example:
+   *   modifyPartCustomProp("part-123", { x: 100, y: 200 });
+   */
+  const modifyPartCustomProp = useCallback(
```

Comment:
We had 3 places that made modifications to part properties that all replicated similar logic. I refactored them out to this helper (and now we have 4 since resizing was added)